sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/UserIdentity.java

```diff
@@ -123,6 +149,21 @@ public Builder setEmail(@Nullable String email) {
       return this;
     }
 
+    /**
+     * Set groups of the user.
+     *
+     * If list of groups is null, no synchronization will be done.
```

Comment:
Javadoc should be completed to explain all cases:
* if `null` then groups are not synchronized. If user does not exist then default group is assigned, else configured groups are unchanged.
* if empty then user groups are reset. User is assigned to default group. (should it be the case ? I don't think it's allowed to have a user without any groups)
* if not empty, then groups are synchronized. What happens if a group does not exist in SQ ? Is it automatically created with default permissions ? Or does it fail ?